src/unknown_0255E0.c

```diff
@@ -367,26 +372,24 @@ void render_scene(Gfx** dList, MatrixS** mtx, Vertex** vtx, s8** tris, s32 updat
             func_80068408(&gSceneCurrDisplayList, &gSceneCurrMatrix);
         }
         gDPPipeSync(gSceneCurrDisplayList++);
-        func_80028CD0(updateRate);
+        initialise_player_viewport_vars(updateRate);
         func_800AB308(-1, -512);
-        if (gCurrentLevelHeader2->weatherEnable > 0 && numViewports < 2) {
-            process_weather(&gSceneCurrDisplayList, &gSceneCurrMatrix, &gSceneCurrVertexList, &gSceneCurrTriList, delta);
+        if (gCurrentLevelHeader2->weatherEnable > 0 && numViewports < VIEWPORTS_COUNT_3_PLAYERS) {
+            process_weather(&gSceneCurrDisplayList, &gSceneCurrMatrix, &gSceneCurrVertexList, &gSceneCurrTriList, tempUpdateRate);
         }
         func_800AD030(get_active_camera_segment());
         func_800ACA20(&gSceneCurrDisplayList, &gSceneCurrMatrix, &gSceneCurrVertexList, get_active_camera_segment());
-        render_hud(&gSceneCurrDisplayList, &gSceneCurrMatrix, &gSceneCurrVertexList, get_racer_object_by_port(D_8011B0B4), updateRate);
+        render_hud(&gSceneCurrDisplayList, &gSceneCurrMatrix, &gSceneCurrVertexList, get_racer_object_by_port(gSceneCurrentPlayerID), updateRate);
     }
-    if ((numViewports == 3) && 
-        (get_current_level_race_type() != RACETYPE_CHALLENGE_EGGS) &&
-        (get_current_level_race_type() != RACETYPE_CHALLENGE_BATTLE) &&
-        (get_current_level_race_type() != RACETYPE_CHALLENGE_BANANAS)) {
-        if (get_multiplayer_hud_setting() == 0) {
+
+    if (numViewports == VIEWPORTS_COUNT_4_PLAYERS && get_current_level_race_type() != RACETYPE_CHALLENGE_EGGS && get_current_level_race_type() != RACETYPE_CHALLENGE_BATTLE && get_current_level_race_type() != RACETYPE_CHALLENGE_BANANAS) {
+        if (get_hud_setting() == 0) {
             if (flip) {
-                gSPSetGeometryMode(gSceneCurrDisplayList++, CVG_X_ALPHA);
+                gSPSetGeometryMode(gSceneCurrDisplayList++, 0x1000);
```

Comment:
Should this be CVG_X_ALPHA or G_CULL_FRONT like above?

Reply:
Ah good catch, that should be `G_CULL_FRONT`

Reply:
yep already fixed